src/reducers/totalBalancesReducer.js

```diff
@@ -73,16 +72,20 @@ export const initialState = {
   isFetching: false,
 };
 
-const setNewBalance = (balancesState, accountId, chain, category, newBalance) => ({
-  ...balancesState,
-  [accountId]: {
-    ...(balancesState?.[accountId] ?? {}),
-    [chain]: {
-      ...(balancesState?.[accountId]?.[chain] ?? {}),
-      [category]: newBalance,
+const setNewBalance = (balancesState, accountId, chain, category, newBalance) => {
+  const accountState = balancesState[accountId] ?? {};
+  const categoryState = accountState[category];
```

Comment:
Would fail on spread if not defined.

```suggestion
  const categoryState = accountState[category] ?? {};
```

Reply:
Applied your suggestion but it seems it will not fail: https://stackoverflow.com/questions/46957194/javascript-es6-spread-operator-on-undefined

Better safe than sorry.